tools/e2e-agent/lvm.go

```diff
@@ -0,0 +1,251 @@
+	Vg                          string `json:"vg"`                          // Volume group
+	ThinPoolAutoExtendThreshold int    `json:"thinPoolAutoExtendThreshold"` // thin pool auto extend threshold
+	ThinPoolAutoExtendPercent   int    `json:"ThinPoolAutoExtendPercent"`   // thin pool auto extend percent
+}
+
+// LvmVersion check lvm version installed on node
+func LvmVersion(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	klog.Info("Get lvm version installed")
+
+	lvmVersionCommand := "lvm version"
+	output, err := bashLocal(lvmVersionCommand)
+	if err != nil {
+		msg = fmt.Sprintf("cannot get lvm version. Error %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrExecFailed, w)
+		return
+	}
+	WrapResult(output, ErrNone, w)
+}
+
+// LvmListVg list lvm vg
+func LvmListVg(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	klog.Info("List lvm vgs")
+
+	lvmListCommand := "vgs --reportformat json"
+	output, err := bashLocal(lvmListCommand)
+	if err != nil {
+		msg = fmt.Sprintf("cannot list lvm vgs Error %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrExecFailed, w)
+		return
+	}
+	WrapResult(output, ErrNone, w)
+}
+
+// LvmListPv list lvm pv
+func LvmListPv(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	klog.Info("List lvm pvs")
+
+	lvmListCommand := "pvs --reportformat json"
+	output, err := bashLocal(lvmListCommand)
+	if err != nil {
+		msg = fmt.Sprintf("cannot list lvm pvs Error %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrExecFailed, w)
+		return
+	}
+	WrapResult(output, ErrNone, w)
+}
+
+// LvmCreatePv create lvm pv
+func LvmCreatePv(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	var lvm Lvm
+	d := json.NewDecoder(r.Body)
+	if err := d.Decode(&lvm); err != nil {
+		msg = fmt.Sprintf("failed to read JSON encoded data, Error: %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrJsonDecode, w)
+		return
+	}
+	if lvm.Pv == "" {
+		msg = "no device path passed"
+		klog.Error(msg)
+		WrapResult(msg, UnprocessableEntityErrorCode, w)
+		return
+	}
+	klog.Info("creates lvm pv, data: %v", lvm)
+
+	lvmPvCreateCommand := fmt.Sprintf("pvcreate %s", lvm.Pv)
+	output, err := bashLocal(lvmPvCreateCommand)
+	if err != nil {
+		msg = fmt.Sprintf("cannot create lvm pv Error %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrExecFailed, w)
+		return
+	}
+	WrapResult(output, ErrNone, w)
+}
+
+// LvmCreateVg create lvm vg
+func LvmCreateVg(w http.ResponseWriter, r *http.Request) {
+	var msg string
+	var lvm Lvm
+	d := json.NewDecoder(r.Body)
+	if err := d.Decode(&lvm); err != nil {
+		msg = fmt.Sprintf("failed to read JSON encoded data, Error: %s", err.Error())
+		klog.Error(msg)
+		WrapResult(msg, ErrJsonDecode, w)
+		return
+	}
+	if lvm.Pv == "" {
+		msg = "no device path passed"
+		klog.Error(msg)
+		WrapResult(msg, UnprocessableEntityErrorCode, w)
+		return
+	} else if lvm.Vg == "" {
```

Comment:
nit: we don't need the else do we?

Reply:
We need to verify vg name otherwise vg creation command will fail

Reply:
I meant it should be an if rather than an else if

Reply:
correct
[fixed](https://github.com/openebs/openebs-e2e/compare/0232e291a02224d4ceb55fe0d1dbd70388c7c935..27cc1a4efba5e2917a1a8596488069251ab82f88)